app/main/controllers/media/RTMediaUserInteraction.php

```diff
@@ -215,8 +215,9 @@ function render() {
 				$icon = "<i class='" . esc_attr( $this->icon_class ) . "'></i>";
 			}
 			$button_start = '<form action="' . esc_url( $link ) . '">';
+			$button_label = apply_filters( 'rtmedia_' . $this->action . '_label_text', esc_html( $this->label ) );
```

Comment:
@surajkrsingh final value of `$button_label` is being escaped before use. Is `esc_html` necessary here ( double escaping ) ?

Reply:
@thrijith  I think it is not required to escape early.

Reply:
Resolved it.